src/nodes/interceptor/eth_sign_transaction.go

```diff
@@ -57,7 +53,14 @@ func (i *Interceptor) ethSignTransaction(ctx context.Context, msg *ethereum.Send
 	if msg.IsPrivate() {
 		sig, err = store.SignPrivate(ctx, msg.From, msg.TxDataQuorum())
 	} else {
-		sig, err = store.SignTransaction(ctx, msg.From, chainID, msg.TxData())
+		var txType int
+		if msg.GasPrice != nil {
```

Comment:
Replace by the pending helper function `msg.isLegacy()` so we have same conditions applied on everywhere

Reply:
done